### PR TITLE
requireCapitalizedComments: improve letter recognition

### LINKED ISSUE #921: JSCS package size has grown significantly since 1.9.0
unicode-6.3.0 was added as part of https://github.com/jscs-dev/node-jscs/pull/847 which is a massive dependency. With 1.8.1 the package was about `7.3Mb` but the newer versions are `127.1Mb`. Is it possible to use a more modular version that only pulls in the category L regexes?

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/7762189-jscs-package-size-has-grown-significantly-since-1-9-0?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mathiasbynens**
Ref. #847. Fixes #921.

cc @alawatthe

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -113,13 +115,14 @@
     "coverage-html": "unit-coverage run -p common -r html -o coverage.html",
     "browserify": "browserify --standalone JscsStringChecker lib/string-checker.js -o jscs-browser.js",
     "changelog": "git log `git describe --tags --abbrev=0`..HEAD --pretty=format:' * %s (%an)' | grep -v 'Merge pull request'",
-    "release": "node publish/prepublish && npm publish",
+    "release": "node tools/generate-patterns.js && npm test && node publish/prepublish && npm publish",
```
Comment (markelog) [inline package.json]: should have done `npm test` at `npm release` from the beginning :-)

Comment (mathiasbynens) [inline package.json]: Well, #YOLO and all that ;)

**Comment by markelog** [comment]
How about we make a module from `tools/generate-pattern.js` by moving it to `publish/helpers` folder and execute it in `publish/prepublish.js`?

**Comment by mathiasbynens** [comment]
@markelog Sounds good. Patch amended.

**Comment by alawatthe** [comment]
Looks good to me. Thanks @mathiasbynens.

I was also fixing #920. But let's merge yours and I will write a separate pull request for #920 once this here is merged.

**Comment by mathiasbynens** [comment]
@alawatthe Alright. Thanks for working on improving Unicode support! :+1:

**Comment by alawatthe** [comment]
It's all because of you :-D

You inspired me with your [talk at jsconf.eu](http://2014.jsconf.eu/speakers/mathias-bynens-javascript-loves-unicode.html). I :heart: it.

**Comment by mathiasbynens** [comment]
That comment just made my day. Thanks!

**Comment by mrjoelkemp** [comment]
Nice work @mathiasbynens. Thanks for the patch. 

@markelog should we fix this up pre-merge to avoid checking in the browser bundle and regexes?

**Comment by mathiasbynens** [comment]
For my projects where I do similar things, I _do_ check in the generated files. It’s useful to see if there are any changes when updating from Unicode v7.0.0 to Unicode v8.0.0 (ETA Q2 2015), for example.

**Comment by mathiasbynens** [comment]
@mrjoelkemp About `jscs-browser.js`, good catch – shall I add it to `.gitignore`?

**Comment by mrjoelkemp** [comment]
@mathiasbynens Agreed about the generated files being checked in. 

I'm not sure why jscs-browser.js wasn't on the gitignore; @markelog would know better.

**Comment by mrjoelkemp** [comment]
I plan on landing this very soon. Sorry it took a while.

**Comment by mrjoelkemp** [comment]
Thanks again @mathiasbynens. Great work!

**Pull request closed without merging**